pkg/capr/planner/planner.go

```diff
@@ -1021,44 +1030,56 @@ func (p *Planner) generatePlanWithConfigFiles(controlPlane *rkev1.RKEControlPlan
 		nodePlan plan.NodePlan
 		err      error
 	)
-	if !controlPlane.Spec.UnmanagedConfig {
-		nodePlan, reg, err = p.commonNodePlan(controlPlane, plan.NodePlan{})
-		if err != nil {
-			return nodePlan, map[string]interface{}{}, "", err
-		}
-		var (
-			joinedServer string
-			config       map[string]interface{}
-		)
-		nodePlan, config, joinedServer, err = p.addConfigFile(nodePlan, controlPlane, entry, tokensSecret, joinServer, reg, renderS3)
-		if err != nil {
-			return nodePlan, config, joinedServer, err
-		}
 
-		nodePlan, err = p.addManifests(nodePlan, controlPlane, entry)
-		if err != nil {
-			return nodePlan, config, joinedServer, err
-		}
+	if controlPlane.Spec.UnmanagedConfig {
+		return plan.NodePlan{}, map[string]interface{}{}, "", nil
+	}
 
-		nodePlan, err = p.addChartConfigs(nodePlan, controlPlane, entry)
-		if err != nil {
-			return nodePlan, config, joinedServer, err
-		}
+	nodePlan, reg, err = p.commonNodePlan(controlPlane, plan.NodePlan{})
+	if err != nil {
+		return nodePlan, map[string]interface{}{}, "", err
+	}
+	var (
+		joinedServer string
+		config       map[string]interface{}
+	)
 
-		nodePlan, err = addOtherFiles(nodePlan, controlPlane, entry)
+	nodePlan, config, joinedServer, err = p.addConfigFile(nodePlan, controlPlane, entry, tokensSecret, joinServer, reg, renderS3)
+	if err != nil {
+		return nodePlan, config, joinedServer, err
+	}
 
-		idempotentScriptFile := plan.File{
-			Content: base64.StdEncoding.EncodeToString([]byte(idempotentActionScript)),
-			Path:    idempotentActionScriptPath(controlPlane),
-			Dynamic: true,
-			Minor:   true,
-		}
+	nodePlan, err = p.addManifests(nodePlan, controlPlane, entry)
+	if err != nil {
+		return nodePlan, config, joinedServer, err
+	}
 
-		nodePlan.Files = append(nodePlan.Files, idempotentScriptFile)
+	mgmtCluster, err := p.managementClusters.Get(controlPlane.Spec.ManagementClusterName)
+	if err != nil {
+		return nodePlan, config, joinedServer, err
+	}
+	if !v3.ClusterConditionBootstrapped.IsTrue(mgmtCluster) {
+		// returning no matter what - even if err because we should end here and don't want the rest of the manifests to get added to the plan if this condition is not set.
+		return nodePlan, config, joinedServer, err
```

Comment:
```suggestion
	if err != nil {
		return nodePlan, config, joinedServer, err
	}
	if !v3.ClusterConditionBootstrapped.IsTrue(mgmtCluster) {
		// exit early as adding remaining manifests before bootstrapping may cause thrashing or operational failures <--- is this more accurate?
		return nodePlan, config, joinedServer, nil
```
AFAICT `err` is always `nil` here.